Snippets/Sqs/Sqs_All/ErrorQueue/ErrorQueue.cs

```diff
@@ -0,0 +1,102 @@
+﻿namespace SqsAll.ErrorQueue
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Amazon.SQS;
+    using Amazon.SQS.Model;
+    using Newtonsoft.Json;
+    using SqsAll.QueueCreation;
+    using Sqs_All;
+
+    public static class ErrorQueue
+    {
+
+        static async Task Usage()
+        {
+            #region sqs-return-to-source-queue-usage
+
+            await ReturnMessageToSourceQueue(
+                    errorQueueName: "error",
+                    messageId: "c390a6fb-4fb5-46da-927d-a156f75739eb")
+                .ConfigureAwait(false);
+
+            #endregion
+        }
+
+        #region sqs-return-to-source-queue
+
+        public static async Task ReturnMessageToSourceQueue(string errorQueueName, string messageId)
+        {
+            var path = QueueNameHelper.GetSqsQueueName($"{errorQueueName}");
+            using (var client = ClientFactory.CreateSqsClient())
+            {
+                var queueUrlResponse = await client.GetQueueUrlAsync(path)
+                    .ConfigureAwait(false);
+                var queueUrl = queueUrlResponse.QueueUrl;
+
+                await InspectMessagesUntilFound(client, messageId, queueUrl)
+                    .ConfigureAwait(false);
+            }
+        }
+
+        static async Task InspectMessagesUntilFound(IAmazonSQS client, string messageId, string queueUrl)
+        {
+            var receivedMessages = await client.ReceiveMessageAsync(new ReceiveMessageRequest
+                {
+                    QueueUrl = queueUrl,
+                    WaitTimeSeconds = 1,
+                    MaxNumberOfMessages = 10,
+                })
+                .ConfigureAwait(false);
+
+            if (receivedMessages.Messages.Count == 0)
+            {
+                return;
```

Comment:
Is this safe - are we sure that if we long-poll for 1 second that we will definitely not miss any messages that may be still in the queue? Would it be safer to long-poll for 20 seconds?